src/handlers/completion.rs

```diff
@@ -197,26 +200,77 @@ fn new_string_arg_completion(value: String) -> CompletionItem {
     }
 }
 
+fn vec_to_completion_list(v: &Vec<String>) -> CompletionList {
+    let items: Vec<CompletionItem> =
+        v.into_iter().map(new_string_arg_completion).collect();
+
+    return CompletionList {
+        is_incomplete: false,
+        items,
+    };
+}
+
+async fn find_compare_completions(
+    params: CompletionParams,
+    ctx: RequestContext,
+    trigger: char,
+) -> Result<CompletionList, String> {
+    let uri = params.text_document.uri;
+    let trigger_len: u32;
+    if trigger == '=' {
+        // != or ==
+        trigger_len = 2
```

Comment:
it feels like these triggers could be pulled out into a type maybe, these feel super magical to me

Reply:
agree, good news is we are about to get our formatter in pretty soon, so things can be must better in situations like this

Reply:
I think its time to get more context from the AST here